### ISSUE TITLE
[Daily Test Failure]  tests/unit/cluster/slot-migration.tcl in test-macos-latest

### ISSUE THREAD

**Issue by PingXie** [madolson/placeholderkv#734]
```
*** [err]: Empty-shard migration target is auto-updated after failover in target shard in tests/unit/cluster/slot-migration.tcl
R 6 didn't assume the replication master in time
*** [err]: Empty-shard migration source is auto-updated after failover in source shard in tests/unit/cluster/slot-migration.tcl
```

https://github.com/valkey-io/valkey/actions/runs/9752760801/job/26916861819#step:5:8406

**Comment by madolson** [comment]
We believe this is a test failure, which is more important to fix before we go GA with Valkey 8.0.

**Comment by PingXie** [comment]
These are caused by race conditions in the tests. The tests in question have been disabled, temporarily. I am going to move this one out of 8.0 project.

**Comment by madolson** [comment]
I think we should keep it in the Valkey 8.0 board. The empty shard failover is a new feature, I think we should be tracking this as a fast follow after the launch.

**Comment by madolson** [comment]
@PingXie Are you still planning on owning this? Maybe we can make sure this is fixed as part of atomic slot migration improvements?

**Comment by PingXie** [comment]
Do we still want to keep the "legacy" non atomic slot migration after we implement atomic slot migration? If not, this test is no longer needed. It is technically a breaking change but I think the impact would be limited to the admin only and I don't see a reason for them to stick with the old protocol.

**Comment by madolson** [comment]
Sounds like a breaking change, so I think we'll need to maintain it for the new future until people adopt the new migration command.

**Comment by PingXie** [comment]
Agreed that this is a breaking change. I am fine with keeping it for a few more release and then reevaluate.
